### PR TITLE
consensus: Groom Logs

### PR DESCRIPTION
**Pull request by alexanderbez**
Executed a local network using simapp and looked for logs that seemed superfluous. This isn't by any means an exhaustive grooming, but should drastically help legibility of logs.

ref: #5912

### CODE REVIEW

**Review by marbar3778 — APPROVED** [review]
Love the reduction in INFO logs. Leaving an approval, but we should wait for others to weigh in

File: consensus/state.go [hunk 1]
```diff
@@ -1474,8 +1474,8 @@ func (cs *State) tryFinalizeCommit(height int64) {
 	if !cs.ProposalBlock.HashesTo(blockID.Hash) {
 		// TODO: this happens every time if we're not a validator (ugly logs)
 		// TODO: ^^ wait, why does it matter that we're a validator?
```
Comment (cmwaters) [inline consensus/state.go:1476]: lol

File: consensus/state.go [hunk 2]
```diff
@@ -1268,7 +1268,7 @@ func (cs *State) enterPrecommit(height int64, round int32) {
 		return
 	}
 
-	logger.Info(fmt.Sprintf("enterPrecommit(%v/%v). Current: %v/%v/%v", height, round, cs.Height, cs.Round, cs.Step))
+	logger.Debug(fmt.Sprintf("enterPrecommit(%v/%v). Current: %v/%v/%v", height, round, cs.Height, cs.Round, cs.Step))
```
Comment (cmwaters) [inline consensus/state.go:1271]: What's your thinking behind having all the `enterPrevote` logs as INFO and `enterPrecommit` as DEBUG?

**Review by cmwaters — COMMENTED** [review]
Yeah I think these changes to DEBUG make sense. I guess the question is what consensus logs deserve to be INFO. From what I've gathered here we want to still show:
1. adding votes and receiving proposals 
2. signing and pushing votes
3. Finalizing commit/block

There are a few other lines where you may want to consider downgrading to DEBUG:
- 924
- 1536
- 1775
- 1818

**Review by erikgrinaker — APPROVED** [review]

File: consensus/state.go (hunk 1 above)
Comment (tessr) [reply consensus/state.go:1476]: good lord

Comment (alexanderbez) [reply consensus/state.go:1476]: I bit there's tons of these tidbits :-p

File: consensus/state.go (hunk 2 above)
Comment (alexanderbez) [reply consensus/state.go:1271]: I missed `enterPrevote`. I'd downgrade that to DEBUG as well.

**Comment by alexanderbez** [comment]
@cmwaters I've downgraded a few more logs per your suggestion. I've also tackled some logs I missed in maverick (sigh, it's really tricky having identical codebases).

WRT to what we still want for INFO level, from what I can tell, I didn't change any of those. Can you confirm?

**Review by cmwaters — APPROVED** [review]
Good work 👍

I had a look at the issue you first wrote up in the Cosmos SDK. Do you think that the default log level in Tendermint, like in the Cosmos SDK should just be "*:info"? I feel the advantage of the granularity of logs is only relevant when debugging (then we focus on the logs of a specific module) but for general use, what is regarded as Info should be from the perspective of the node, not of the individual modules.

Side note: I'm well aware that the maverick node is a bit of an annoyance to update. Once we are settled on the reactor pattern, I plan to simplify it so that we won't need to update it every time we make a change to consensus.

**Comment by erikgrinaker** [comment]
> Do you think that the default log level in Tendermint, like in the Cosmos SDK should just be "*:info"? I feel the advantage of the granularity of logs is only relevant when debugging (then we focus on the logs of a specific module) but for general use, what is regarded as INFO should be from the perspective of the node, not of the individual modules.

I agree with this. Users are frequently confused by apparent halts because we don't log any progress information e.g. during fast sync. I'd go even further and say that `INFO` should be from the perspective of what the typical user is interested in. If we don't feel like this is granular enough, we could consider adding a `NOTICE` level (see syslog levels) for slightly more important stuff and default to that, but I'd avoid that unless necessary.

**Comment by tessr** [comment]
I wonder if we should also include documentation somewhere on how to filter logs by level. 

(I'm imagining the Discord conversation where we try to help a user debug something, and we're asking for more exhaustive logs.)

**Comment by marbar3778** [comment]
> I wonder if we should also include documentation somewhere on how to filter logs by level.
> 
> (I'm imagining the Discord conversation where we try to help a user debug something, and we're asking for more exhaustive logs.)

we do have a section here https://docs.tendermint.com/master/tendermint-core/running-in-production.html#logging that also leads to https://blog.cosmos.network/one-of-the-exciting-new-features-in-0-10-0-release-is-smart-log-level-flag-e2506b4ab756. Ill transfer the blog article to our docs for better visibility. 

Generally agree, we should go with `*:info` and if people are annoyed they can change it. Better to show too much then get asked the same question in discord 20 times.

**Comment by cmwaters** [comment]
> I wonder if we should also include documentation somewhere on how to filter logs by level.

Sometimes, when I don't know something git or golang related I will use `git help ....` or `go help ...` which usually provide quite extensive descriptions on how something works. Perhaps we should consider beefing up some of Tendermint's (and also the SDK's) help commands in a similar manner.

**Comment by alexanderbez** [comment]
Ok, so based off of out-of-band discussions, the current logging infrastructure provides us what already need:

- Support for standard levels (`info|warn|fatal|debug|...)`
- Support for granular levels (`[module]:[level]`)

What we need to do is simply change the _default_ level to `info` and provide better documentation for super users (which @marbar3778 is working on).

**Comment by marbar3778** [comment]
>     * Support for standard levels `info|warn|fatal|debug`

To not confuse anyone reading through this PR, Tendermint supports `info|error|debug`.

**Review by melekes — APPROVED** [review]
👍

**Pull request merged**